src/Sylius/Component/Promotion/Processor/PromotionProcessor.php

```diff
@@ -60,24 +60,20 @@ public function process(PromotionSubjectInterface $subject)
             $this->promotionApplicator->revert($subject, $promotion);
         }
 
-        $eligiblePromotions = [];
+        $preQualifiedPromotions = $this->preQualifiedPromotionsProvider->getPromotions($subject);
 
-        foreach ($this->preQualifiedPromotionsProvider->getPromotions($subject) as $promotion) {
-            if (!$this->promotionEligibilityChecker->isEligible($subject, $promotion)) {
-                continue;
-            }
-
-            if ($promotion->isExclusive()) {
+        foreach ($preQualifiedPromotions as $promotion) {
+            if ($this->promotionEligibilityChecker->isEligible($subject, $promotion) && $promotion->isExclusive()) {
```

Comment:
`$promotion->isExclusive()` costs less than `$this->promotionEligibilityChecker->isEligible($subject, $promotion)` so it should be executed first.